pkg/controller/controller_test.go

```diff
@@ -242,6 +242,60 @@ func TestAddIngressWithNameAnnotationShouldCreateMonitorAndDelete(t *testing.T)
 	checkMonitorWithName(t, monitorName, false)
 }
 
+func TestUpdateIngressNameAnnotationShouldUpdateMonitorAndDelete(t *testing.T) {
+	namespace := "test"
+	url := "google.com"
+	ingressName := "name-annotation-ingress"
+
+	controller := getControllerWithNamespace(namespace, true)
+
+	stop := make(chan struct{})
+	defer close(stop)
+	go controller.Run(1, stop)
+
+	ingress := util.CreateIngressObject(ingressName, namespace, url)
+
+	ingress = addMonitorAnnotationToIngress(ingress, true)
+
+	monitorName := "monitor-friendly-name"
+	ingress = addMonitorNameAnnotationToIngress(ingress, monitorName)
+
+	result, err := controller.kubeClient.ExtensionsV1beta1().Ingresses(namespace).Create(ingress)
+	log.Printf("Created ingress %q.\n", result.GetObjectMeta().GetName())
+
+	updatedMonitorName := "monitor-friendly-name-updated"
+	ingress = addMonitorNameAnnotationToIngress(ingress, updatedMonitorName)
+	result, err = controller.kubeClient.ExtensionsV1beta1().Ingresses(namespace).Update(ingress)
+
+	if err != nil {
+		panic(err)
+	}
+	log.Printf("Updated ingress %q.\n", result.GetObjectMeta().GetName())
+
+	time.Sleep(5 * time.Second)
+
+	// Old monitor name can be used to fetch the new monitor
+	checkMonitorWithName(t, monitorName, true)
```

Comment:
Why is it so? That old monitor name can be used to fetch the new monitor?

Reply:
Ah! I thought it was a weird quirk of their API that the old name could be used for the new monitor, or perhaps there was a caching layer in the IMC code that I didn't understand.

But, when I looked more closely at GetByName, it searches by keyword and the old monitor name `monitor-friendly-name` was a substring of the new one `monitor-friendly-name-updated`, so it was showing up for that reason. I'll change the test to use completely separate strings so this doesn't happen.

Reply:
You just found out a bug in uptimerobot's implementation :joy: The GetByName method just returns the 1st element of the array fetched from the API. Instead it should iterate through the array and return the exact match. Its a short fix. Can you do that as well while you're at it? Would be great!

Reply:
Oh, that makes sense too. will do!